scripts/retry-execution.sh

```diff
@@ -0,0 +1,120 @@
+#!/bin/bash
+
+FAILED_STATUS="ExecutionFailed"
+NOT_FOUND_STATUS="NOT_FOUND"
+
+function get_attempt_number {
+  local arn=$1
+  local attempt=$(echo ${arn} | perl -lne 'print $1 if /^.*-R(\d{2})$/')
+
+  echo $attempt
+}
+
+function get_next_execution_arn {
+  local arn=$1
+  local attempt=$(get_attempt_number ${arn})
+  if [[ -z "${attempt}" ]]
+  then
+    local state_machine_arn_part=$(echo ${arn} | perl -lne 'print $1 if /^(.*):.*$/')
+    local execution_name_part=$(echo ${arn} | perl -lne 'print $1 if /^.*:(.*)$/')
+
+    # State machine execution name has 80 characters limit
+    echo "${state_machine_arn_part}:${execution_name_part:0:76}-R01"
+    return
+  fi
+
+  local next_attempt=$(perl -e "print sprintf(\"%02d\", \"${attempt}\" + 1)")
+
+  local arn_without_attempt_number=$(echo ${arn} | perl -lne 'print $1 if /^(.*-R)\d{2}$/')
+  echo "${arn_without_attempt_number}${next_attempt}"
+}
+
+function get_execution_name {
+  local arn=$1
+  local execution_name=$(echo ${arn} | perl -lne 'print $1 if /^.*:(.*)$/')
+
+  echo $execution_name
+}
+
+function load_execution_details {
+  local arn=$1
+  local execution_history=$(aws stepfunctions get-execution-history --reverse-order --execution-arn ${arn})
+  local execution_status=$(echo "${execution_history}" | jq -r '.events[0] | .type')
```

Comment:
Where is this script intended to be run out of curiosity?

Reply:
We manually run this script on our local machines to re-run failed state machine executions in production.